module/User/src/Model/User.php

```diff
@@ -152,13 +152,34 @@ public function getRoleNames()
     }
 
     /**
-     * Get the user's role ID.
-     *
      * @return string
      */
-    public function getRoleId()
+    public function getRoleId(): string
     {
-        return 'user_' . $this->getLidnr();
+        $roleNames = $this->getRoleNames();
+        if (in_array('admin', $roleNames)) {
+            return 'admin';
+        }
+
+        if (in_array('company_admin', $roleNames)) {
+            return 'company_admin';
+        }
+
+        if (empty($roleNames)) {
+            return 'user';
+        }
```

Comment:
A user who is an active member (and does not have an `admin` or `company_admin` role) does not get the `active_member` role, as this will evaluate to true and return. I suggest moving it below the active member check.

Reply:
Not sure why it's in this spot indeed. I'll move it.